app/Http/Controllers/PublicProfileController.php

```diff
@@ -29,6 +29,23 @@ public function index()
             ->with('speakers', $users);
     }
 
+    public function search(Request $request)
+    {
+        $users = User::search($request->search_query)
```

Comment:
Let's rename this `query` and access it using `$request->get('query')` just for consistency.